frappe/api/v2.py

```diff
@@ -144,7 +169,10 @@ def execute_doc_method(doctype: str, name: str, method: str | None = None):
 	doc.is_whitelisted(method)
 
 	doc.check_permission(PERMISSION_MAP[frappe.request.method])
-	return doc.run_method(method, **frappe.form_dict)
+	result = doc.run_method(method, **frappe.form_dict)
+	doc.reload()
```

Comment:
This doesn't seem necessary. Is it being done to undo changes that aren't commited to DB? If so, those bugs are rare and this cost of reloading document is going to be there always.